int64[2] start
int64[2] end
---
int64[] path

